### DISCUSSION TITLE
Nuxt 3 - Socket hang up - Error?

### DISCUSSION THREAD

**Discussion by n-u-l-l-e-d** [nuxt/nuxt#17065] [Questions]
I get "socket hang up" error when calling an asynchronous function after writing to a file. 

``` vue
<script setup lang="ts">
import winston from "winston";

if (process.server) {
  const logger = winston.createLogger({
    format: winston.format.json(),
    transports: [
      new winston.transports.File({ filename: "info.log", level: "info" }),
    ],
  });

  const simulateRequest = (ms) => {
    return new Promise((resolve) => setTimeout(resolve, ms));
  };

  const successTest = async () => {
    await simulateRequest(500);
    logger.info("successTest");
    console.log("successTest");
  };

  const errorTest = async () => {
    logger.info("errorTest");
    await simulateRequest(500); // <----- SOCKET HANG UP
    console.log("errorTest");
  };

  await successTest();
  await errorTest();
}
</script>
```
``` console
ERROR  [proxy] socket hang up

  at connResetException (node:internal/errors:691:14)
  at Socket.socketOnEnd (node:_http_client:471:23)
  at Socket.emit (node:events:402:35)
  at Socket.emit (node:domain:475:12)
  at endReadableNT (node:internal/streams/readable:1343:12)
  at processTicksAndRejections (node:internal/process/task_queues:83:21)
```

It also occurs when writing files (`fs.writeFile`) using an asynchronous request through the API. I used Winston for the example to show it more easily:

[Nuxt 3 - CodeSandbox](https://codesandbox.io/s/nuxt3-fps8k?file=/pages/index.vue:0-713) (Node 16)

If I replicate the code directly in Express it does not generate any errors: 

[Express - CodeSandbox](https://codesandbox.io/s/express-smoyz?file=/src/index.js) (Node 16)

**Comment by steklopod** [comment]
Sometimes I also have same error  log with `node 17` and  latest version of `"@nuxt/bridge": "npm:@nuxt/bridge-edge@^3.0.0-27306529.1ef6e3f"`. But not everytime

**Comment by polly3d** [comment]
During my develop, I got this error when calling an api in `server/api` folder after saving anything.

**Reply by lucpotage** [reply]
Same for me.

**Comment by multiplehats** [comment]
Can confirm!

**Comment by awacode21** [comment]
I regularly get this error while running npm run dev, working on the project and saving my files.

**Reply by wc-davide** [reply]
same

**Reply by KeionneDerousselle** [reply]
same

**Reply by KeionneDerousselle** [reply]
Just noticed that it stops when I'm not running Jest in watch mode at the same time.

**Comment by Pflyg** [comment]
I get this error every few minutes as well. I use the boilerplate nuxt-bridge configuration. No special api calls or anything

**Reply by ThomasBerne** [reply]
same here.

**Reply by kon3ko** [reply]
same

**Comment by CodyBontecou** [comment]
I also get this regularly. Tends to require me to restart my local dev server.

Let me know if there's a log output or something I can contribute to solving this.

**Comment by gustavonikov** [comment]
I get twice this error when I'm using https server config and refresh the page :/. Pretty similar to https://github.com/nuxt/nuxt.js/issues/4324 issue.
![image](https://user-images.githubusercontent.com/68268899/154495022-1bb7bfa3-0bbe-46be-8e45-950fb7c8cb02.png)

**Comment by sharkfin009** [comment]
Me too! am running Algolia and Strapi modules.

**Reply by sharkfin009** [reply]
problem has gone away after removing windicss and going back to tailwind

**Reply by multiplehats** [reply]
I'm running an app on Nuxt 3 with Windicss and a different app with Tailwind and can confirm that this issue happens with both

**Reply by wc-davide** [reply]
I don't think it's related to any module... I have the same issue and I don't use any.

**Comment by devlsh** [comment]
Seem to be having the same issue, but more constant, when making GraphQL requests on a page with `prefetch: true` (HTTP request runs on the server)

**Comment by jd-solanki** [comment]
I am using lerna and still getting this error

**Comment by anburocky3** [comment]
I'm also facing the same problem. I have to do re-run inorder to avoid this error.

**Comment by pavloskafritsas** [comment]
I had the same issue but I solved it by changing the node version from 18.x to the latest LTS (16.15.1).

**Comment by wesley3295** [comment]
Think this is a OS issue with windows. If using windows try running on a virtual OS. ex)WSL

**Reply by Pflyg** [reply]
No, I've had this happen on Linux as well

**Comment by asrorbek-mannonov** [comment]
I'm using MacOS with node version v20.8.0. I can confirm that it happens frequently while `pnpm run dev`
